elements/rh-icon/test/rh-icon.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { expect, html, oneEvent } from '@open-wc/testing';
+import { createFixture } from '@patternfly/pfe-tools/test/create-fixture.js';
+import { RhIcon } from '@rhds/elements/rh-icon/rh-icon.js';
+
+describe('<rh-icon>', function() {
+  let element: RhIcon;
+  beforeEach(async function() {
+    element = await createFixture(html`<rh-icon></rh-icon>`);
+  });
+
+  describe('simply instantiating', function() {
+    it('imperatively instantiates', function() {
+      expect(document.createElement('rh-icon')).to.be.an.instanceof(RhIcon);
+    });
+
+    it('should upgrade', async function() {
+      const klass = customElements.get('rh-icon');
+      expect(element)
+          .to.be.an.instanceOf(klass)
+          .and
+          .to.be.an.instanceOf(RhIcon);
+    });
+  });
+
+  describe('when the icon has a fallback content', function() {
```

Comment:
a11ySnapshot assertions for role (can wait until after pfe-tools from select typeahead is released)